### PR TITLE
Consolidate .gitignore files under the root.

### PR DESCRIPTION
**Pull request by jgraham**
Multiple .gitignores aren't supported by the manifest update code when running
against the filesystem, so put all the rules in the root file instead.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,11 +24,25 @@ node_modules/
...
 .vscode/
-.DS_Store
-*.rej
+\#*
+scratch
+
+# Testsuite-specific rules
+/conformance-checkers/vnu.jar
+/cors/resources/log.txt
+/css//build-temp
```
Comment (gsnedders) [inline .gitignore]: Probably shouldn't have `//` here?

File: .gitignore [hunk 2]
```diff
@@ -22,11 +24,25 @@ node_modules/
...
+/css/dist
+/css/dist_last
+/css/tools/cache
+/webaudio/idl/*
```
Comment (gsnedders) [inline .gitignore]: Probably should have trailing new line. (Also, do we actually want `/*` at the end here? That's very surprising.)

**Review by gsnedders — CHANGES_REQUESTED** [review]
If we're going to do this, we should add a lint to avoid new `.gitignore` files being added.

File: .gitignore (hunk 2 above)
Comment (jgraham) [reply .gitignore]: I literally just copied the existing rule. I'm not sure what the point of it actually is.

**Comment by gsnedders** [comment]
@jgraham can you rebase this so the lint unit tests run?

**Comment by jgraham** [comment]
Done

**Review by gsnedders — APPROVED** [review]

**Pull request merged**